### ISSUE TITLE
:farmer: Gz-transport11-homebrew `UNIT_gz_TEST ignTest.TopicInfo` failed

### ISSUE THREAD

**Issue by Crola1702** [gazebosim/gz-transport#510]
## Environment
* OS Version: MacOS
* Source build

## Steps to reproduce
1. Run a build in [gz-transport11](https://build.osrfoundation.org/job/gz_transport-ci-ign-transport11-homebrew-amd64/)
2. See `ignTest.TopicInfo` fail

## Output

Reference build: https://build.osrfoundation.org/job/gz_transport-ci-ign-transport11-homebrew-amd64/59/consoleFull

Log output:
```
33: [ RUN      ] ignTest.TopicInfo
33: /Users/jenkins/jenkins-agent/workspace/gz_transport-ci-ign-transport11-homebrew-amd64/gz-transport/src/cmd/gz_TEST.cc:135: Failure
33: Value of: infoFound
33:   Actual: false
33: Expected: true
33: OUTPUT[No publishers on topic [/foo]
33: ] Size[30]. Expected Size=50
33: 
33: /Users/jenkins/jenkins-agent/workspace/gz_transport-ci-ign-transport11-homebrew-amd64/gz-transport/src/cmd/gz_TEST.cc:138: Failure
33: Value of: output.find("ignition.msgs.Vector3d") != std::string::npos
33:   Actual: false
33: Expected: true
33: [  FAILED  ] ignTest.TopicInfo (26598 ms)
```

Might be related to #509

**Comment by scpeters** [comment]
I suspect that this is a flaky test. It is now failing consistently on macOS since [build 58](https://build.osrfoundation.org/job/gz_transport-ci-ign-transport11-homebrew-amd64/58/), though it still passes on Ubuntu. There were no changes in the `ign-transport11` branch from [build 57](https://build.osrfoundation.org/job/gz_transport-ci-ign-transport11-homebrew-amd64/57/) to [build 58](https://build.osrfoundation.org/job/gz_transport-ci-ign-transport11-homebrew-amd64/58/), and I compared the log files and did not see any significant changes in dependencies. Looking at the [code for this failing test case](https://github.com/gazebosim/gz-transport/blob/b0328a184ca52227b70286f437333a0afc0baf02/src/cmd/gz_TEST.cc#L109-L142), I see it starts a publisher and waits a few seconds, which seems like a race condition to me.

**Comment by azeey** [comment]
Fixed by #551.
